.github/workflows/pull-request-with-code.yml

```diff
@@ -17,35 +17,62 @@ on:
       - '**/*.toml'
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
 env:
-  ORG_GRADLE_PROJECT_signingKey : ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGKEY }}
+  ORG_GRADLE_PROJECT_signingKey: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGKEY }}
   ORG_GRADLE_PROJECT_signingKeyId: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGKEYID }}
-  ORG_GRADLE_PROJECT_signingKeyPassword : ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGPASSWORD }}
+  ORG_GRADLE_PROJECT_signingKeyPassword: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGPASSWORD }}
   CLI_TEST_MAX_DURATION_IN_SECONDS: 10
 
 # Note that "jobs.build.strategy" and "jobs.build.runs-on" should be kept in sync with "pull-request-wth-code"
 jobs:
   build:
     strategy:
       matrix:
-        os: [ ubuntu-latest, windows-latest ]
-        # When changing the list of JDK versions, the build configuration has to be changed by a repository admin. See
-        # https://github.com/pinterest/ktlint/pull/1787#issuecomment-1409074092
-        jdk: [ 8, 11, 17, 19 ]
-        exclude: # windows with JDK8 are *really* flaky
-          - os: windows-latest
-            jdk: 8
+        os: [ubuntu-latest, windows-latest]
     runs-on: ${{ matrix.os }}
+    name: "[build] OS: ${{ matrix.os }}"
     steps:
       - uses: actions/checkout@v3
+
       - uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
-          java-version: ${{ matrix.jdk }}
+          java-version: 20 # This should match all `jvmToolchain(xxx)` calls in the project
+
       - uses: gradle/gradle-build-action@v2
         with:
           gradle-home-cache-cleanup: true
+
       - name: Build with release Kotlin version
         run: ./gradlew build ktlintCheck --no-configuration-cache
-      - name: Build with dev Kotlin version
+
+      - name: Build with assemble Kotlin version
         run: ./gradlew -PkotlinDev build ktlintCheck --no-configuration-cache
+
+  tests:
+    strategy:
+      fail-fast: false
```

Comment:
🆕 New thing, I can drop it if you don't like it. 
Previously, if one of the branches failed the remaining ongoing ones were cancelled. With `fail-fast: false` the workflow will continue running, so one will now know if the failure affects single OS x JDK configuration of i.e. all Windows builds are affected

Reply:
I think it is fine to add this. It might take a couple of minutes more before the build fails as the Windows builds are considerably slower than the Ubuntu builds.

One thing that would be very nice is if the `ktlintCheck` is run on all modules before the tests are executed.

Reply:
>One thing that would be very nice is if the ktlintCheck is run on all modules before the tests are executed.

Do I understand correctly, you'd imagine having a
```
step 1: ktlintCheck
step 2 (in parallel)
	- build + build kotlinDev
	- testOnJdk8
	- testOnJdk11
    - testOnJdk17
```
so to basically defer both build and tests until `ktlintCheck` completes? Or did you mean something else in mind? 

❔ Additional question: Should I move `kotlinDev` to a separate job as well? 

I'll open another PR targeting this one, to have clear diff, pl to avoid updating branch protection settings more than once 👀

Reply:
Yes, lets discuss that in the other PR. But you understood the gist of my idea. I am not sure what moving `kotlinDev` to a separate job would entail and do no see any direct benefit from it. I will leave that up to you to decide.

Reply:
I'll mark this discussion as resolved, I pushed another set of changes and commented them out: https://github.com/pinterest/ktlint/pull/2120#discussion_r1264650925, let's continue the discussion there :)